adam/learner/objects.py

```diff
@@ -520,7 +512,7 @@ def _keep_hypothesis(  # pylint: disable=arguments-differ
         hypothesis: PerceptionGraphTemplate,
         bound_surface_template: SurfaceTemplateBoundToSemanticNodes,  # pylint:disable=unused-argument
     ) -> bool:
-        if len(hypothesis.graph_pattern) < 2:
+        if len(hypothesis.graph_pattern) < 3:
```

Comment:
Document this change to explain why 3 is now our bound -- The comment seems to imply 2.